python/pyspark/sql/connect/functions/builtin.py

```diff
@@ -4027,25 +4028,210 @@ def make_timestamp_ntz(
     mins: "ColumnOrName",
     secs: "ColumnOrName",
 ) -> Column:
-    return _invoke_function_over_columns(
-        "make_timestamp_ntz", years, months, days, hours, mins, secs
+    ...
+
+
+@overload
+def make_timestamp_ntz(
+    *,
+    years: "ColumnOrName",
+    months: "ColumnOrName",
+    days: "ColumnOrName",
+    hours: "ColumnOrName",
+    mins: "ColumnOrName",
+    secs: "ColumnOrName",
+) -> Column:
+    ...
+
+
+@overload
+def make_timestamp_ntz(
+    *,
+    date: "ColumnOrName",
+    time: "ColumnOrName",
+) -> Column:
+    ...
+
+
+def make_timestamp_ntz(  # type: ignore[misc]
+    *args: "ColumnOrName",
```

Comment:
shall we remove the `*args: "ColumnOrName"` and only support following patterns? @HyukjinKwon 

```
make_timestamp_ntz(years=y, months=mo, days=d, hours=h, mins=mi, secs=s)
make_timestamp_ntz(years=y, months=mo, d, h, mi, s)
make_timestamp_ntz(y, mo, d, h, mi, s)

make_timestamp_ntz(date=d, time=t) <- for the new code path
```

Reply:
If we keep `*args: "ColumnOrName"`, we need a helper function to preprocess the arguments so that it can be reused in all the places.

`def _preprocess_make_timestampe_args(...) -> Tuple[ColumnOrName, ..., ColumnOrName]` which returns a 8-tuple for the 8 arguments